### DISCUSSION TITLE
How do I use raw Xrgb values?

### DISCUSSION THREAD

**Discussion by xbjfk** [image-rs/image#2062] [Q&A] [answered]
Hello, I am writing a program that receives Xrgb values. By that, I mean that there are 4 8 bit channels, x, r, g and b, where x is an unused padding channel. What would be the easiest way to manipulate this kind of image using `from_raw`?
Thanks.

**Comment by fintelia** [comment, marked as answer]
The easiest thing is probably to do a pass over your image and convert it into the format expected by `from_raw` (packed R, G, and B channels in that order with no padding).

**Comment by xbjfk** [comment]
Right, so something like? (pseudocode):
```
let rgb_pixels = vec<u8>();
foreach pixel in xrgb_pixels<u32> {
    rgb_pixels.add([
        pixel >> 16 & 0xFF,
        pixel >> 8 & 0xFF,
        pixel >> 0 & 0xFF,
    ])
}
```

**Reply by xbjfk** [reply]
Argh, I was worried about doing this because of inefficiency, but it's either doing this now or when encoding - because to my knowledge no image format uses XRGB. Maybe this gives me a good excuse to play with simd :)
